pdl-compiler/src/backends/rust_legacy/serializer.rs

```diff
@@ -143,15 +143,18 @@ impl<'a> FieldSerializer<'a> {
         let shift = self.shift;
 
         match &field.desc {
-            ast::FieldDesc::Flag { optional_field_id, set_value, .. } => {
-                let optional_field_id = optional_field_id.to_ident();
+            ast::FieldDesc::Flag { optional_field_ids, .. } => {
+                assert!(optional_field_ids.len() == 1, "condition flag reuse not supported by legacy generator");
+
+                let (cond_field_id, set_value) = &optional_field_ids[0];
```

Comment:
nit: `let (optional_field_id, set_value) = &optional_field_ids[0];` to limit the diff